src/types.js

```diff
@@ -137,12 +137,15 @@ export type CollectionsListOptions = {
   userId?: string
 };
 
-export type AccessToken = ?string | ShareDescriptor;
+export type AccessToken = ?string;
 export type AccessTokenOption =
   | AccessToken // TODO: Deprecate?
   | (() => AccessToken) // TODO: Deprecate
   | (() => Promise<AccessToken>);
 
+export type ShareIdentifier = ?string | ShareDescriptor;
```

Comment:
I think the maybe type here will allow `shareId: null`

```suggestion
export type ShareIdentifier = string | ShareDescriptor;
```

Reply:
We actually `null` out `shareToken` in ui. So when we call `getShareToken`, it could technically be `null`. If we make this change then I guess I would just pass in an empty string if `shareToken` is `null`?

Reply:
@anthony-j-castro that makes sense. In that case we could omit the `shareId` option when `getShareToken` is null

Can cast null to undefined with `getShareToken() || undefined` or can update getShareToken to return `undefined` initially. Since getShareToken is synchronous, If it's `undefined` it won't ever be defined

> I would just pass in an empty string if shareToken is null?

Personally I avoid `""` outside of a default value for user input. It seems to introduce invalid states that make it harder to reason about control flow - as you lose the ability to distinguish between the states: `string | null | undefined`

 With `""`, null and undefined are collapsed into `string` via `""` and that is likely not the expected string so `string` loses meaning too